### ISSUE TITLE
r.import fails with temporary region [Bug]

### ISSUE THREAD

**Issue by baharmon** [OSGeo/grass#595]
**Description of Bug**
r.import fails when using a temporary region in a Python script.

**To Reproduce**
Run this python script after adding a path to directory with .tif files to import.
I used this [Modeled Historic NLCD dataset](https://www.sciencebase.gov/catalog/item/59d3c73de4b05fe04cc3d1d1).

```
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
import grass.script as gscript

# settings
gscript.use_temp_region()
env = gscript.gisenv()
overwrite = True
env['GRASS_OVERWRITE'] = overwrite
env['GRASS_VERBOSE'] = False
env['GRASS_MESSAGE_FORMAT'] = 'standard'
gisdbase = env['GISDBASE']
location = env['LOCATION_NAME']
mapset = env['MAPSET']

# set paths
directory = os.path.normpath(r"") # <--- ADD PATH TO DIRECTORY WITH TIF FILES

# iterate through files in directory
for (dirpath, dirnames, filenames) in os.walk(directory):
    # iterate through tif rasters
    for file in filenames:
        if file.endswith('.tif'):
            # import raster
            gscript.run_command('r.import',
                input=os.path.join(directory, file),
                output=os.path.splitext(file)[0],
                flags='e',
                overwrite=overwrite)
        else:
            pass
```

**Expected Behavior**
This script should import and reproject .tif rasters in a directory. It works after removing 
`gscript.use_temp_region()`.

**Error Message**
```
ERROR: Unable to open element file <windows> for <tmp.reproduce_rimport_bug.py.40280@PERMANENT>
ERROR: Unable to read GDAL dataset <D:\gisdata\ModeledHistoricNCLD_1938_1992\CONUS_Backcasting_y1938.tif>
Traceback (most recent call last):
  File
"C:\Users\baharmon\Downloads\reproduce_rimport_bug.py", line
31, in <module>
    overwrite=overwrite)
  File "C:\Program Files\GRASS GIS
7.8\etc\python\grass\script\core.py", line 441, in
run_command
    return handle_errors(returncode, returncode, args,
kwargs)
  File "C:\Program Files\GRASS GIS
7.8\etc\python\grass\script\core.py", line 343, in
handle_errors
    returncode=returncode)
grass.exceptions.CalledModuleError: Module run None r.import
--o -e input=D:\gisdata\ModeledHistoricNCLD_1938_1992\CONUS_
Backcasting_y1938.tif output=CONUS_Backcasting_y1938 ended
with error
Process ended with non-zero return code 1. See errors in the
(error) output.
```

**System description**
- Operating System: Windows 10
- GRASS GIS version 7.8.2

**Comment by metzm** [comment]
You can not use `use_temp_region()` in a python script that calls another python script also using `use_temp_region()` because this will overwrite the environment variable `WIND_OVERRIDE`. `use_temp_region()` should fail with a fatal error if `WIND_OVERRIDE` is already defined.

**Comment by metzm** [comment]
I found the problem:

Script 1 can set a temp region and then call script 2, which will use the temp region of script 1.
Script 2 can set its own temp region, which is deleted when script 2 exits, and the original env var `"WIND_OVERRIDE"` of script 1 is restored, such that script 1 can continue to use its own temp region. See tests in the closed PR #630.

Here, `r.import` creates a new temporary location for import. This new location does not contain the temp region defined by `"WIND_OVERRIDE"`. This temp region does not make sense anyway in the new location because the new location uses a different CRS.

A fix would be to add a test to `r.import` when it creates a new location, that the env vars `"WIND_OVERRIDE"`and `"GRASS_REGION"` must not be set.

Other modules/scripts creating temporary mapsets or locations will experience the same problem.
